src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
-public class SalaryInfo {
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class SalaryInfo extends CheckNotNull {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        checkNotNull(names, data, dateFrom, dateTo);
+        String[] fullInformation = new String[names.length];
+        StringBuilder builder;
+        StringBuilder finalBuilder = new StringBuilder();
+        SimpleDateFormat format = new SimpleDateFormat("dd.MM.yyyy");
+        short builderCounter = 0;
+        for (String name : names) {
+            int sum = 0;
+            builder = new StringBuilder();
+            for (String note : data) {
+                if (!note.isEmpty()) {
+                    try {
+                        String[] dataSeparated = note.split(" ");
+                        Date parsedNoteDate = format.parse(dataSeparated[0]);
+                        Date parsedDateFrom = format.parse(dateFrom);
+                        Date parsedDateTo = format.parse(dateTo);
```

Comment:
Please, feel free to move parsing of `dateFrom` and `dateTo` outside of the loop, as these values are not being changed.